File: starterkits/javascript/client/LocalGameClient.js
const { GameMessage } = require("../GameInterface");

class LocalGameClient {
  /**
   * @param {Solver} solver
   */
  constructor(solver) {
    console.log("[Running in local mode]");
    const sample = new GameMessage({
      tick: 1,
      payload: {
        totems: [{ shape: "I" }],
      },
    });
    solver.getAnswer(sample);
  }
}

module.exports = { LocalGameClient };

File: starterkits/rust/src/client/local_game_client.rs
use crate::solver::Solver;
use crate::game_interface::{GameMessage, Question, Totem, TotemQuestion};

pub struct LocalGameClient {
    solver: Solver,
}

impl LocalGameClient {
    pub fn new(solver: Solver) -> Self {
        LocalGameClient { solver }
    }

    pub async fn run(&self) {
        println!("[Running in local mode]");

        let totem_question = TotemQuestion { shape: Totem::I };
        let question = Question { totems: vec![totem_question] };
        let game_message = GameMessage { tick: 1, payload: question };

        self.solver.get_answer(&game_message).expect("There was an error in the solver's code!");
    }
}
